### PR TITLE
GCP logging/monitoring configurable

### PR DESCRIPTION
**Pull request by contil**
## what
* GCP logging and monitoring for container now configurable in the module
* cleanup examples and add the oslogin settings for the child module

## why
* GCP monitoring and logging are optional
* oslogin can be useful if we need to login into the VM / container

### CODE REVIEW

**Review by kpocius — APPROVED** [review]
LGTM, but since it introduces a new metadata attribute, this becomes a breaking change as it will re-create the instance.

Maybe we can use this opportunity and flip oslogin to false by default, what do you think, @bschaatsbergen?

**Comment by contil** [comment]
@kpocius I see oslogin being set to false already by default in the variable file .... where do you see it set to true?

**Comment by kpocius** [comment]
@contil, sorry for the confusion. My memory tricked me -- I thought it was set to true in order not to become a breaking change. Turns out it wasn't.

**Comment by bschaatsbergen** [comment]
@contil thanks for raising this PR, I'll go through it tomorrow 👍🏼

File: examples/secure-env-vars/main.tf [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-  managed_zone          = "<your-managed-zone>"
+  project_id                   = "<your-project-id>"
+  network                      = "<your-network>"
+  subnetwork                   = "<your-subnetwork>"
+  region                       = "<your-region>"
+  zone                         = "<your-zone>"
+  image                        = "<your-image>"
+  domain                       = "<example.com>"
+  managed_zone                 = "<your-managed-zone>"
+  enable_oslogin               = false
```
Comment (bschaatsbergen) [inline examples/secure-env-vars/main.tf:10]: Can we remove the locals from all the examples, as these are optional variables anyways and now add extra entries to the list in the example.

File: examples/complete/main.tf [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-  domain       = "<example.com>"
-  managed_zone = "<your-managed-zone>"
+  project_id                   = "<your-project-id>"
+  region                       = "<your-region>"
+  zone                         = "<your-zone>"
+  domain                       = "<example.com>"
+  managed_zone                 = "<your-managed-zone>"
+  enable_oslogin               = false
+  google_logging_enabled       = true
+  google_logging_use_fluentbit = false
```
Comment (bschaatsbergen) [inline examples/complete/main.tf:9]: > Can we remove the locals from all the examples, as these are optional variables anyways and now add extra entries to the list in the example.

File: examples/basic/main.tf [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+  project_id                   = "<your-project-id>"
+  network                      = "<your-network>"
+  subnetwork                   = "<your-subnetwork>"
+  region                       = "<your-region>"
+  zone                         = "<your-zone>"
+  domain                       = "<example.com>"
+  managed_zone                 = "<your-managed-zone>"
+  enable_oslogin               = false
+  google_logging_enabled       = true
+  google_logging_use_fluentbit = false
```
Comment (bschaatsbergen) [inline examples/basic/main.tf:11]: > Can we remove the locals from all the examples, as these are optional variables anyways and now add extra entries to the list in the example.

File: variables.tf [hunk 4]
```diff
@@ -139,3 +139,21 @@ variable "labels" {
   description = "Key-value pairs representing labels attaching to instance & instance template"
   default     = {}
 }
+
+variable "google_logging_enabled" {
+  type        = bool
+  description = "Enable Google Cloud Logging"
```
Comment (bschaatsbergen) [inline variables.tf:145]: What does it exactly enable? As logging already works, if I look a Cloud Logging I can already see container logs being written there.

File: variables.tf [hunk 5]
```diff
@@ -139,3 +139,21 @@ variable "labels" {
...
+
+variable "google_logging_use_fluentbit" {
+  type        = bool
+  description = "Enable Google Cloud Logging using Fluent Bit ( available from COS 105+ )"
+  default     = false
+}
+
+variable "google_monitoring_enabled" {
+  type        = bool
+  description = "Enable Google Cloud Monitoring"
```
Comment (bschaatsbergen) [inline variables.tf:157]: What does it exactly enable?

**Review by bschaatsbergen — CHANGES_REQUESTED** [review]
Thanks for this PR, it looks ok - though I'm curious what it exactly enables as I already see logs and metrics.. can you be a bit more specific on the metadata attributes please? :)

File: examples/secure-env-vars/main.tf (hunk 1 above)
Comment (contil) [reply examples/secure-env-vars/main.tf:10]: the main reason I created this PR was to allow to make these settings configurable ... i.e. logging is being transitioned from old to new ( using now fluentbit etc ). The main idea was to make sure we have terraform clearly state the requirements and do not depend on defaults as well for different versions of COS etc.

File: examples/complete/main.tf (hunk 2 above)
Comment (contil) [reply examples/complete/main.tf:9]: I added the locals to make sure the user understand the choices are deliberate and can be changed ONLY looking at the locals section without altering the bulk of the code past the locals.

File: variables.tf (hunk 4 above)
Comment (contil) [reply variables.tf:145]: Logging is being changed and these settings should allow to make some choices that will enforce settings even when defaults will change and/or to disable some features

File: variables.tf (hunk 5 above)
Comment (contil) [reply variables.tf:157]: metadata settings and their mapping to COS milestones are described in https://cloud.google.com/container-optimized-os/docs/how-to/logging

**Comment by bschaatsbergen** [comment]
Hi @contil, I was on holiday for a long period - I'm addressing some changes this week and will look into yours!

**Comment by bschaatsbergen** [comment]
@contil your proposed change is introduced in #126

**Pull request closed without merging**